crates/chain/src/local_chain.rs

```diff
@@ -634,7 +646,7 @@ where
         match (curr_orig.as_ref(), curr_update.as_ref()) {
             // Update block that doesn't exist in the original chain
             (o, Some(u)) if Some(u.height()) > o.map(|o| o.height()) => {
-                changeset.blocks.insert(u.height(), Some(u.data()));
+                changeset.blocks.insert(u.height(), u.data());
```

Comment:
This is incorrect. If we reach a placeholder checkpoint, we don't want to insert anything into the changeset. Since a value of `None` in the changeset means deletion (which we are not doing).